packages/amplify-cli/src/execution-manager.ts

```diff
@@ -0,0 +1,125 @@
+        await executePluginModuleCommand(context, pluginModule);
+    }
+}
+
+async function executePluginModuleCommand(context: Context, plugin: PluginInfo) {
+    const { commands, commandAliases } = plugin.manifest;
+    if (!commands!.includes(context.input.command!)) {
+        context.input.command = commandAliases![context.input.command!];
+    }
+
+    if(fs.existsSync(plugin.packageLocation)){
+        const pluginModule = require(plugin.packageLocation);
+        await raisePreEvent(context); 
+        await pluginModule.executeAmplifyCommand(context);
+        await raisePostEvent(context); 
+    }else{
+        await scan(); 
+        context.print.error('The Amplify CLI plugin platform detected an error.');
+        context.print.info('It has performed a fresh scan.'); 
+        context.print.info('Please execute your command again.');
+    }
+}
+
+async function raisePreEvent(context: Context){
+    if(context.input.plugin === Constant.CORE){
+        if(context.input.command === 'init'){
+            await raisePreInitEvent(context); 
+        }else if(context.input.command === 'push'){
+            await raisePrePushEvent(context); 
+        }
+    }
+}
+
+async function raisePreInitEvent(context: Context){
+    await raiseEvent(context, new AmplifyEventArgs(
+        AmplifyEvent.PreInit,
+        new AmplifyPreInitEventData()
+    )); 
+}
+
+async function raisePrePushEvent(context: Context){
+    await raiseEvent(context, new AmplifyEventArgs(
+        AmplifyEvent.PrePush,
+        new AmplifyPrePushEventData()
+    )); 
+}
+
+async function raisePostEvent(context: Context){ 
+    if(context.input.plugin === Constant.CORE){
+        if(context.input.command === 'init'){
+            await raisePostInitEvent(context); 
+        }else if(context.input.command === 'push'){
+            await raisePostPushEvent(context); 
+        }
+    }
+}
+
+async function raisePostInitEvent(context: Context){
+    await raiseEvent(context, new AmplifyEventArgs(
+        AmplifyEvent.PostInit,
+        new AmplifyPostPushEventData()
+    )); 
+}
+
+async function raisePostPushEvent(context: Context){ 
+    await raiseEvent(context, new AmplifyEventArgs(
+        AmplifyEvent.PostPush,
+        new AmplifyPostInitEventData()
+    )); 
+}
+
+export async function raiseEvent(context: Context, args: AmplifyEventArgs){
+    const plugins = getPluginsWithEventHandler(context.pluginPlatform, args.event);
+    if(plugins.length > 0){
+        const sequential = require('promise-sequential'); 
+        const eventHandlers = plugins.filter((plugin)=>{
+            return fs.existsSync(plugin.packageLocation);
+        }).map((plugin)=>{
+            return async () => {
+                try{
+                    const pluginModule = require(plugin.packageLocation);
+                    await pluginModule.handleAmplifyEvent(context, args);
+                }catch{
+                    //no need to need anything
```

Comment:
Because individual plugin's event handler can throw error, and we don't want one plugin's malfunction to cause the event broadcasting to error out. so we wrap each individual event handler inside a try block.